gravatar/src/main/java/com/gravatar/services/AvatarService.kt

```diff
@@ -22,51 +16,35 @@ public class AvatarService(private val okHttpClient: OkHttpClient? = null) {
         const val LOG_TAG = "AvatarService"
     }
 
-    // Run onResponse and onError callbacks on the main thread
-    private val coroutineScope = CoroutineScope(GravatarSdkDI.dispatcherMain)
-
     /**
      * Uploads a Gravatar image for the given email address.
      *
      * @param file The image file to upload
      * @param email The email address to associate the image with
      * @param wordpressBearerToken The bearer token for the user's WordPress/Gravatar account
-     * @param gravatarUploadListener The listener to notify of the upload result
      */
-    public fun upload(
-        file: File,
-        email: Email,
-        wordpressBearerToken: String,
-        gravatarUploadListener: GravatarListener<Unit, ErrorType>,
-    ) {
+    public suspend fun upload(file: File, email: Email, wordpressBearerToken: String): Result<Unit, ErrorType> {
         val service = GravatarSdkDI.getGravatarApiV1Service(okHttpClient)
```

Comment:
I know this is not part of your changes but I think it's best if this is defined as a class property instead of a local val as it seems for every time the `upload` function is called, it  setups the underlying retrofit service. I guess we could avoid doing that as I don't see anything changing upon call of the `upload` function that would require recreating the retrofit service. But yeah I stand corrected if I'm missing something and hence it was designed this way.

Reply:
Nice suggestion! 🚀